Source/EasyNetQ/Consumer/Pipeline.cs

```diff
@@ -0,0 +1,103 @@
+using EasyNetQ.DI;
+using EasyNetQ.Interception;
+using EasyNetQ.Logging;
+
+namespace EasyNetQ.Consumer;
+
+public readonly record struct ConsumeContext(
+    in MessageReceivedInfo ReceivedInfo,
+    in MessageProperties Properties,
+    in ReadOnlyMemory<byte> Body,
+    in CancellationToken CancellationToken
+);
+
+public delegate ValueTask<AckStrategy> ConsumeDelegate(ConsumeContext context);
+
+public interface IConsumeMiddleware
+{
+    ValueTask<AckStrategy> InvokeAsync(ConsumeContext context, ConsumeDelegate next);
```

Comment:
```suggestion
    ValueTask<AckStrategy> InvokeAsync(in ConsumeContext context, ConsumeDelegate next);
```

Is this possible? Async restrictions?

Reply:
It is possible to declare it as in here, but in most implementations InvokeAsync will be declared as async, because next should be awaited.

Reply:
ok